### ISSUE TITLE
Sanity check for output

### ISSUE THREAD

**Issue by sebastianpos** [homalg-project/CAP_project#191]
Should we implement a **sanity check** for the output of
- basic operations
- functors

that checks whether the output really matches the output type
inferable from the

- `io_type` of basic operations,
- the `Range` category of the functor?

If yes, should this sanity check be **optional** or **non-optional**?
If optional, should the default be **activated** or **deactivated**?

In my opinion, a sanity check that complies with the previous CAP philosophy
(which states that the user is responsible for matching the specifications)
should be optional and deactivated.
As indicated by #188, however, having a sanity check for output might
be a nice debugging tool.

**Comment by mohamed-barakat** [comment]
Does **optional** mean that such a sanity check need not be implemented? If yes then I am in favour of non-optional + deactivated. So if debugging is needed one can easily switch in on.

**Comment by mohamed-barakat** [comment]
Thank you, then I agree with you and vote for **optional** + **deactivated**.

**Comment by sebasguts** [comment]
Indeed, those checks should be deactivated. We could have two levels of checking again:

1. Check output type from `io_type`: This can be completely derived from the `io_type` of a basic operation, and for a functor (and natural transformation) this is always the same, as partially given in #188.

2. Check diagram form, i.e., does a kernel embedding composed with the input is zero. This has to be included by hand.

Both checks should be optional, and clearly not given by the user, but in the method record. We could give the possibility for the user to add a third-level of checking by a user-given function. Although I do not see any benefit of this.
